File: tests/audio/soundrecorder.cpp
#include <iostream>
#include <SFML/Audio.hpp>

class DerivableSoundRecorder : public sf::SoundRecorder
{
public :
	DerivableSoundRecorder();

protected:
	virtual bool onStart();
	virtual bool onProcessSamples(const sf::Int16* samples, std::size_t sampleCount);
	virtual void onStop();
};

DerivableSoundRecorder::DerivableSoundRecorder():
sf::SoundRecorder ()
{
};

bool DerivableSoundRecorder::onStart()
{
	std::cout << "onStart()" << std::endl;
	return true;
}

bool DerivableSoundRecorder::onProcessSamples(const sf::Int16* samples, std::size_t sampleCount)
{
	std::cout << "onProcessSamples()" << std::endl;
	return true;
}

void DerivableSoundRecorder::onStop()
{
	std::cout << "onStop()" << std::endl;
}

File: tests/audio/soundrecorder.py
import sfml as sf

class MySoundRecorder(sf.SoundRecorder):
	def __init__(self):
		sf.SoundRecorder.__init__(self)

	def on_start(self):
		print("on_start")
		return True
		
	def on_process_samples(self, chunk):
		print("on_process_samples")
		print(type(chunk))
		return True
		
	def on_stop(self):
		print("on_stop")
		
foo = MySoundRecorder()

input()
foo.start(44100)
input("a")
print("b")
foo.stop()
input("foo")
input("bar")
